tests/test_operations.py

```diff
@@ -303,127 +304,154 @@ def test_secrets_crud(self):
+                )
+                with open(tempfile, 'r') as f:
+                    file_contents = f.read()
+                self.assertIn(testvar_value, file_contents)
+
+                # Output file already exists, use force
+                with self.assertRaises(RuntimeError):
                     secrets.get_secret(
-                        [],
-                        argparse.Namespace(secret_names=[testvar_name], json=True),
+                        [], argparse.Namespace(secret_name=testvar_name, outfile=tempfile, force=False)
                     )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
-                # Multiple variables:
+                secrets.get_secret(
+                    [], argparse.Namespace(secret_name=testvar_name, outfile=tempfile, force=True)
+                )
+
+                # Stdout
                 with CaptureStdout() as output:
                     secrets.get_secret(
-                        [],
-                        argparse.Namespace(
-                            secret_names=[testvar_name, unusedvar_name],
-                            json=True
-                        )
+                        [], argparse.Namespace(secret_name=testvar_name, outfile=None, force=False)
                     )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
+                self.assertIn(testvar_value, "\n".join(output))
+
+                # Clean up
+                subprocess.call(['rm', '-f', tempfile])
 
         with self.subTest("Update existing secret"):
-            with mock.patch("dss.operations.secrets.secretsmanager") as sm:
+            with mock.patch("dss.operations.secrets.sm_client") as sm:
                 # Updating the variable will try to get secret value and succeed
                 sm.get_secret_value = mock.MagicMock(return_value={"SecretString": testvar_value})
                 # Next we will call the update secret command
                 sm.update_secret = mock.MagicMock(return_value=None)
-                # Update secret
+
+                # Update secret:
+                # Dry run first
+                with SwapStdin(testvar_value2):
+                    secrets.set_secret(
+                        [],
+                        argparse.Namespace(secret_name=testvar_name, dry_run=True, infile=None, force=True),
+                    )
+
+                # Use stdin
+                with SwapStdin(testvar_value2):
+                    secrets.set_secret(
+                        [],
+                        argparse.Namespace(secret_name=testvar_name, dry_run=False, infile=None, force=True),
+                    )
+
+                # Use input file
+                tempfile = '.dss-test-operations-temp-input'
```

Comment:
You might like `tempfile.NamedTemporaryFile`.  It creates a tempfile in a context manager and deletes it upon exit, for example:
```
with tempfile.NamedTemporaryFile(prefix='.dss-test-operations-temp-input') as temp_file:
    temp_file.write(testvar_value2)
    secrets.set_secret([], argparse.Namespace(secret_name=testvar_name, dry_run=False, infile=temp_file.name, force=True))
```

Reply:
Great suggestion